packages/inventory/src/components/filters/useTagsFilter.js

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect } from 'react';
+import { constructGroups, mapGroups } from '../../shared';
+import { Spinner } from '@patternfly/react-core/dist/esm/components/Spinner';
+
+export const tagsFilterState = { tagsFilter: {} };
+export const TAGS_FILTER = 'TAGS_FILTER';
+export const tagsFilterReducer = (_state, { type, payload }) => ({
+    ...type === TAGS_FILTER && {
+        tagsFilter: payload
+    }
+});
+
+export const useTagsFilter = (
+    allTags = [],
+    loaded = false,
+    additionalTagsCount = 0,
+    onShowMoreClick,
+    [ globalState, dispatch ] = [ tagsFilterState ]
```

Comment:
I feel a bit unsure about this line, could you explain it to me?

Reply:
Sure thing! It's pretty easy, it destructs array into 2 items and if no array is passed it will use defaul state so nothing breaks.

The call can look something like
```JS
const allTags = ['something'];
const allTagsLoaded = false;
const additionalTagsCount = 0;
const onToggleClicked = () => console.log('I was clicked!');
const state = {
  some: 'value'
};
const stateDispatch = () => console.log('dispatch was called!');
useTagsFilter(allTags, allTagsLoaded, additionalTagsCount, () => dispatch(toggleTagModal(true)), [ state, stateDispatch ]);
```